app/ca.go

```diff
@@ -0,0 +1,117 @@
+package app
+
+import (
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+)
+
+type caCert struct {
+	Fingerprint       string    `json:"fingerprint"`
+	Issuer            string    `json:"issuer"`
+	Subject           string    `json:"subject"`
+	NotBefore         time.Time `json:"notBefore"`
+	NotAfter          time.Time `json:"notAfter"`
+	Base64EncodedData string    `json:"base64EncodedData"`
+}
+
+type caCerts []caCert
+
+func parseCertificates(encodedCerts string) (caCerts, error) {
+
+	// decode the cert bundle
+	bundle, err := base64.StdEncoding.DecodeString(encodedCerts)
+	if err != nil {
+		return nil, err
+	}
+	if len(bundle) == 0 {
+		return caCerts{}, nil
+	}
+
+	certBlocks := [][]byte{}
+	var blocks []byte
+	p := bundle
+	for {
+		var block *pem.Block
+		var rem []byte
+		block, rem = pem.Decode(p)
+		if block == nil {
+			break
+		}
+		blocks = append(blocks, block.Bytes...)
+		certBlocks = append(certBlocks, []byte(strings.TrimSpace(string(p[:len(p)-len(rem)]))))
+		p = rem
+	}
+	if len(blocks) == 0 {
+		return nil, fmt.Errorf("failed to decode certificates")
+	}
+	// If p is greater than 0, then this means that there was a portion of the certificate that
+	// is/was malformed.
+	if len(p) > 0 {
+		return nil, fmt.Errorf("failed to parse one or more certificates, remainingBytesLen=%d", len(p))
+	}
+	certs, err := x509.ParseCertificates(blocks)
```

Comment:
this data comes from the server, right? it's fine to be safe, but am wondering when we might hit this situation

Reply:
This code path is used to parse certs coming from the server as well as certs provided by the user.